rfcs/0000-changes-lifecyle.md

```diff
@@ -0,0 +1,205 @@
+- Feature Name: (fill me in with a unique ident, `my_awesome_feature`)
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [FuelLabs/sway-rfcs#0000](https://github.com/FuelLabs/sway-rfcs/pull/001)
+- Sway Issue: [FueLabs/sway#0000](https://github.com/FuelLabs/sway/issues/001)
+
+# Summary
+
+[summary]: #summary
+
+This RFC intends to propose a scheme to track changes in the compiler. 
+
+# Motivation
+
+[motivation]: #motivation
+
+The compiler needs a guide on how to introduce changes to improve developer experience every time a compiler is updated.
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+## PR changes
+
+Every PR will have a section named "Release Notes", which will contains a user friendly explanation of what the PR contains.
+
+The compiler release notes **WILL** be the amalgamation of these section "as is".
+
```

Comment:
I also think that we should discuss formalizing what our policy towards breaking changes is here.

Some of this is getting discussed elsewhere for other moving parts of the ecosystem, but I think we should commit to not making breaking changes outside of specific timed releases and emergencies (security related bugs, etc).

The normal process should be that a breaking changes goes through behind a feature flag and those features get stabilized at predictable intervals when we do a breaking release.